### DISCUSSION TITLE
Changing the width of the main container?

### DISCUSSION THREAD

**Discussion by Thanatermesis** [pacollins/calligraphy#3] [Ideas]
I asked recently how to make bigger the width of the main container:

![image](https://user-images.githubusercontent.com/1181626/163105808-c0d698f3-bd39-487b-8bf2-f2c4307e96ff.png)

And @pacollins recently explained me that:

> The container width is set based on [optimal line length](https://baymard.com/blog/line-length-readability) and is not currently adjustable. Filling the screen will actually lead to UX problems on larger screens without being careful. At this point, I have no plans to change this on the theme. If you look at popular sites that focus on reading/writing content (like Medium), they do not fill all available space for the same reason.

This makes a lot of sense to me and is likely the best way to read articles, on the other hand side I feel that my (long) articles will be quite short in width and very long scrolling to read them, especially the ones related to programming which may require more amount of text.

Another reason for wanting it bigger can be the need to put contents in more specific ways (like, one image on the left and text on the right of it), especially using asciidoc contents.

Could be good if we can define the width of the articles, or maybe a front-matter setting to select between "static" and "dynamic" (dynamic would be something like filling the empty side spaces) because different types of articles may be wanted to be set in different ways. In any case if this can be defined in the source code could be good to have on it a comment with the previous link that explains the importance of the default optimal line length which is very interesting and an important setting to consider.

**Comment by pacollins** [comment]
> especially the ones related to programming which may require more amount of text.

From a UX standpoint, it is hard for our eyes to track long lines of text without getting lost. To do that effectively, we would need to increase line height. Not to say we couldn't. If you really wanted to take up more horizontal space, the next best option would be to allow for multiple columns of text. This doesn't really make sense in a web format though.

Maybe an example would help me better understand the use case you see?

![image](https://user-images.githubusercontent.com/11359433/163678360-2d129fa8-f383-49e5-a047-f203c54320ca.png)

> Another reason for wanting it bigger can be the need to put contents in more specific ways (like, one image on the left and text on the right of it), especially using asciidoc contents.

I do think that this is a good argument for allowing expanded width. I remember seeing a layout once where images and other resources were placed to the side of the main content (where the authors and table of contents is for us). I wonder if this as a shortcode would be a good solution.

On an unrelated note, which browser are you using in this screenshot? The table of contents is not displaying properly.

**Reply by Thanatermesis** [reply]
One of the ideas to create a blog for myself is to make public some writings that I have since years, these writings were originally in pure plain text so I wanted to move them to asciidoc which is a good way to reformat them (more featured contents, better-structured contents like image-on-left-of-text, etc, can be converted to many good looking formats, etc...), that's why Hugo was a good option to use, just put my writings there and make the public :), so an option to have a wider main-content could be good to have, as some 

In fact, the idea of a shortcode (front-matter?) makes more sense, the ability to use a wider content based on the needs of each page, maybe a line assigning the value? for example:
* main-content-width: default
* main-content-width: full
* main-content-width: 100%
* main-content-width: 75%

----
> On an unrelated note, which browser are you using in this screenshot? The table of contents is not displaying properly.

Its chrome on inspect mode + scaled-down-50%, but don't worry about that, the TOC does works correctly (dynamically following), I just tried it again, in the screenshot it was just a few pixels before it triggers to the next section

**Reply by pacollins** [reply]
[Shortcode](https://gohugo.io/content-management/shortcodes/) are content, whereas [front matter](https://gohugo.io/content-management/front-matter/) is more data.

I'll think on this some more.

**Reply by Thanatermesis** [reply]
I don't know much yet about Hugo but the ideal will be that one thing doesn't interfere with the other, more specifically, if my articles are written in asciidoc will be good that placing them in the hugo tree I can render them as a website, or at the same time i can render an article (or book) individually with commands like "asciidoctor-epub3 article.adoc -o book.epub" without having errors of contents that I need to remove to make it working, but as I said, I have not played with this enough to see how to do these things :)

**Comment by pacollins** [comment]
After thinking about this for a while, I think the best solution is to utilize [a grid like Google Fonts used in their redesign](https://design.google/library/reimagining-google-fonts/) or Bootstrap. Then we could allow for some degree of customization by adding additional columns.

**Reply by pacollins** [reply]
I ended up scraping the 12-column grid, and here is what I am at now.

https://jsbin.com/bisatolaju/edit?css,output

This would give the flexibility to define `--content-width` via `config.toml`. The only real issue I see is that allowing `--sidebar-width` to be modify destroys the breakpoints. In an ideal world, the sidebars are a minimum width, so we could calculate `sidebar-width + content-min-width` to provide the breakpoints via a SCSS variable.

Thoughts @Thanatermesis ?

**Comment by Thanatermesis** [comment]
That looks pretty good and dynamic at the same time :)

Sidebars are not so important to define, is better if they are auto, only the "content" is a one that the user (author) would like to define
